src/main/java/bob/command/Command.java

```diff
@@ -0,0 +1,40 @@
+package bob.command;
+
+import bob.exception.BobException;
+import bob.exception.BobInvalidTaskNumberException;
+import bob.storage.StorageFile;
+import bob.task.TaskList;
+import bob.ui.TextUi;
+
+/**
+ * A command encapsulates logic that can be executed to interact with the application.
+ * It can be stored an executed at a later instance or be reused.
+ */
+public abstract class Command {
+    /**
+     * Executes the current command and can modify the current state of the task list or
+     * print out messages.
+     *
+     * @param taskList Current list of tasks
+     * @param storageFile Storage file reader and writer
+     * @param ui User Interface reader and writer
+     * @throws BobException if the command is constructed with invalid arguments
+     */
```

Comment:
Overall, I think the Javadocs are done well, and the descriptions provided are easy to understand and provide the important information for the class/methods.